src/cryptonote_core/service_node_list.h

```diff
@@ -776,8 +868,13 @@ class service_node_list {
         std::vector<key_image_blacklist_entry> key_image_blacklist;
         std::unordered_map<crypto::x25519_public_key, crypto::public_key> x25519_map;
         block_height height{0};
-        mutable quorum_manager quorums;  // Mutable because we are allowed to (and need to) change
-                                         // it via std::set iterator
+        // Mutable because we are allowed to (and need to) change it via std::set iterator:
+        mutable quorum_manager quorums;
+        // blockchaindb-global-transaction-index => confirmation metadata of unconfirmed L2 state
+        // changes.  This is an *ordered* map because confirmation votes in a pulse block depend on
+        // the order of txes in here.
+        std::map<crypto::hash, unconfirmed_l2_tx> unconfirmed_l2_txes;
```

Comment:
Is the blockchaindb-global-transaction-index the 32 byte hash, because the comment on `l2_votes` says it's a monotonic tx index?

```
    // vector of L2 state changes for L2 state change transactions that are still unconfirmed in
    // this block.  true = vote for confirmation, false = vote for rejection.  Votes are sorted in
    // blockchain order of unconfirmed transactions (i.e. by the monotonic internal transaction
    // index).
    std::vector<bool> l2_votes;
```

If it were the hash there'd be an issue with the mapping of block to unconfirmed txs in a block since they'd get sorted depending on the hash in the SNL.

Reply:
> Is the blockchaindb-global-transaction-index the 32 byte hash, because the comment on l2_votes says it's a monotonic tx index?

Oops, leftover comment: I was trying to use the global tx index, but there is apparently no way to get a tx out of the database by global index, which is why I switched to hash.

It *does* change the sort order because now they are sorted by hash, but that's okay because, as you said, it only matters within that single block.

Reply:
Pushed a fix for the misleading comment.